pkg/client/restclient/request.go

```diff
@@ -105,16 +106,13 @@ type Request struct {
 	resource     string
 	resourceName string
 	subresource  string
-	selector     labels.Selector
 	timeout      time.Duration
 
 	// output
 	err  error
 	body io.Reader
 
-	// The constructed request and the response
-	req  *http.Request
-	resp *http.Response
+	ctx context.Context
```

Comment:
Currently the plan is only timeouts for e2es. Timeouts, deadlines, and cancellation are all very powerful tools for clients, and `context.Context` is emerging as a standard Go way to support these. Also, it has been built directly into the `http` package.

In the future, if we start passing `context.Context` throughout more of our code, it could facilitate per-request local tracing, per-request credentials, and other things.